pkg/l4lb/l4netlbcontroller.go

```diff
@@ -655,7 +656,51 @@ func (lc *L4NetLBController) shouldUseNEGBackends(service *v1.Service) bool {
 	if utils.HasL4NetLBFinalizerV3(service) {
 		return true
 	}
-	return lc.enableNEGAsDefault
+	if lc.enableNEGAsDefault {
+		// Do an extra check in case the customer removes the required finalizers
+		// Get the backend service if one exists and see what kind of backend
+		// is attached. If it's an IG keep using IGs.
+		backendType, err := lc.backendTypeOfTheBackendService(service, svcLogger)
+		if err != nil {
+			svcLogger.Error(err, "Could not determine the backend type of the existing backend service, deferring to default")
+			return true
+		}
+		if backendType != nil && *backendType == instanceGroupLink {
+			return false
+		}
+		return true
```

Comment:
If in the future there would be something different then NEG or IG this statement will return false and we will try provision NEGs. 

```suggestion
		return backendType == nil || *backendType == negGroupLink
```

Reply:
done